iterators task (1)/student/include/impl.h

```diff
@@ -2,20 +2,77 @@
 #include <iterator>
 
 template <class Container>
-struct accumulator
+struct accumulator 
 {
-    /** @todo Iterator traits*/
-    /** @todo Constructor with container*/
-    /** @todo Copy assignment operator*/
-    /** @todo Move assignment operator*/
+private:
+    Container& mContainer;
+public:
+    /** Iterator traits*/
+    using iterator_category = std::output_iterator_tag;
+    using value_type =  typename Container::value_type;
+    using difference_type = typename Container::difference_type;
+    using pointer = typename Container::pointer;
+    using reference = typename Container::reference;
```

Comment:
```suggestion
    using reference = void; 
```

https://stackoverflow.com/questions/2619551/output-iterators-value-type

Reply:
Changed to void also for difference_type and pointer